content/blog/eng/common-problems-faced-by-php-developers-in-consuming-an-ampq-message.md

```diff
@@ -0,0 +1,298 @@
+---
+authors: ["emulator000"]
+date: "2018-11-15"
+draft: true
+share: true
+categories: [English, RabbitMQ, AMQP, Consumer, PHP, Rust, Go]
+title: "Common problems faced by PHP developers in consuming an AMQP message"
+type: "post"
+languageCode: "en-EN"
+toc: true
+---
+
+When we use RabbitMQ and our project is in PHP, we have to run a Consumer and we could encounter some common and still unresolved problems.
+
+In order to implement RabbitMQ in the project, there are different ready to use packages, especially when we use a framework like Symfony or Laravel. Usually these packages provides the full needed integration for the AMQP protocol so, we can easily configure it and create a Producer, a Consumer and then, we have to create a little script that will supervisor the PHP process in order to make sure that all our consumers are running. 
+
+# Consumer
+
+In order to consume a message, we have to run a demon that will listen for new messages and then, process each message executing some logic; this is a normal flow in case of blocking requests, we don't want to block the user request nor loose it.
```

Comment:
```suggestion
In order to consume a message, we have to run a demon that will listen for new messages, then process each message executing some logic: this is a normal flow in case of blocking requests, and we don't want to block the user request nor loose it.
```